File: leap_year.rb
# http://rosettacode.org/wiki/Leap_year

def leap_year?(year)
  case
  when year % 400 == 0
    return true
  when year % 4 == 0 && year % 400 ==0
    return true
  else
    return false
  end      
end  

years_to_test = [1700,1800,1900,2100,2200,1600,2000,2400,1994,1996,1997,1998,1999]

years_to_test.each do |year|
  puts "#{year} is a leap year? #{leap_year?(year)}"
end  

File: leap_year.swift
// http://rosettacode.org/wiki/Leap_year

func leapYear(year: Int) -> Bool {
  if (year % 400 == 0) {
    return true
  } else if (year % 4 == 0 && year % 100 != 0) {
    return true
  } else {
    return false
  }
}

println("2000: \(leapYear(2000))")
println("1900: \(leapYear(1900))")
println("1800: \(leapYear(1800))")
println("1996: \(leapYear(1996))")
println("1999: \(leapYear(1999))")

